.eslintrc

```diff
@@ -24,6 +47,7 @@
         "react-hooks/rules-of-hooks": "error",
         "react-hooks/exhaustive-deps": "warn",
         "consistent-return": "off",
+        "react/destructuring-assignment": ["warn", "never"],
```

Comment:
why do we need to disable destructuring 😕

Reply:
I think it'd be good if we adopted one pattern and stick with it. There can be exceptions tho, that's why it is just a warning and there are cases where I ignored the ESLint warning.

There are legitimate cases to use it, but in general, I think it's easier to see what variables come from the `props` if you always have `props.myVariable`.

Reply:
As long as the rule is not forcing us to write myVariable as  `props.myVariable` it's fine

Reply:
> As long as the rule is not forcing us to write myVariable as props.myVariable it's fine

++

Reply:
> But isn't destructuring the props pattern being widely used by everyone after react functional components.

It seems a little old way to write myVariable as props.myVariable 🥲

I don't know about that. Do you have any reference proving that prop destructuring is being widely used? I believe that having `props` is easier to read as if I refer `props.myVariable` lets me know that this variable comes from props instantly.

If you have cases like `({ id })` and id is a prop, it gets harder to know where this prop came from in big Components.

I'm open to enabling destructuring props, but then we should adopt one approach. Having mixed Components using `props` or `({ id, prop2 ... })` honestly creates confusion.

Reply:
Not to mention that by having many props or Components with long names the props will be broken down when destructuring which also kinda (my personal feel) becomes a code smell issue.

Reply:
> If you have cases like ({ id }) and id is a prop, it gets harder to know where this prop came from in big Components.

You can see where the props came from because we are using typescript and interfaces. 

```
const AnchoredHeading = ({ children, level, id }: AnchorHeadingProps) => {...
```
The above example tells you exactly where the props came from which is `AnchorHeadingProps`

> I don't know about that. Do you have any reference proving that prop destructuring is being widely used?

[AirBnB's ](https://github.com/airbnb/javascript/blob/5c01a1094986c4dd50a6ee4d9f7617abdfabb58a/README.md#destructuring)style guide prefer's destructuring and that is a very popular style guide for typescript/javascript

Reply:
Interesting. Thanks for the headsup, @benhalverson. I think that's fair argument, so I'm going to change the setting to always request destructuring.